src/server/origin-address/validator.js

```diff
@@ -0,0 +1,92 @@
+export default (/** @type {OriginAddress} */ originAddress) => {
+  const { addressLine1, addressTown, addressPostcode } = originAddress
+
+  let valid = true
+  const errors = {}
+
+  if (!addressLine1) {
+    valid = false
+    errors.addressLine1 = {
+      text: 'Enter address line 1, typically the building and street'
+    }
+  } else if (!validateLength(addressLine1)) {
+    valid = false
+    errors.addressLine1 = {
+      text: 'Address line 1 must be no longer than 255 characters'
+    }
+  }
+
+  if (!validateLength(originAddress?.addressLine2, false)) {
+    valid = false
+    errors.addressLine1 = {
+      text: 'Address line 2 must be no longer than 255 characters'
+    }
+  }
+
+  if (!addressTown) {
+    valid = false
+    errors.addressTown = {
+      text: 'Enter town or city'
+    }
+  } else if (!validateLength(originAddress.addressTown)) {
+    valid = false
+    errors.addressLine1 = {
+      text: 'Address line 2 must be no longer than 255 characters'
+    }
+  }
+
+  if (!addressPostcode) {
+    valid = false
+    errors.addressPostcode = {
+      text: 'Enter postcode'
+    }
+  } else if (!validatePostcode(originAddress.addressPostcode)) {
+    valid = false
+    errors.addressPostcode = {
+      text: 'Enter a full UK postcode'
+    }
+  }
+
+  return [valid, errors]
+}
```

Comment:
I'd like to side-by-side this with a Joi-based implementation (which is the lib CDP typically uses for validation).  I'll test that in the background & share with